packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx

```diff
@@ -148,13 +151,38 @@ export const ChatListItem = ({ chat, myId, isSelected, isRead }: IMgtChatListIte
           return loadChatWithPreview(graph, id);
         };
         load(chatInternal.id).then(
-          c => setChatInternal(c),
+          c => {
+            setChatInternal(c);
+            checkWhetherToMarkAsRead(c);
+          },
           e => error(e)
         );
       }
     }
   }, [chatInternal]);
 
+  // check whether to mark the chat as read or not
+  const checkWhetherToMarkAsRead = (c: Chat) => {
+    cache.loadLastReadTime(c.id!).then(lastReadData => {
+      if (lastReadData) {
+        const lastUpdatedDateTime = new Date(c.lastUpdatedDateTime as string);
+        const lastMessagePreviewCreatedDateTime = new Date(c.lastMessagePreview?.createdDateTime as string);
+        const lastReadTime = new Date(lastReadData.lastReadTime as string);
+        if (
+          lastUpdatedDateTime > lastReadTime ||
+          lastMessagePreviewCreatedDateTime > lastReadTime ||
+          lastReadData.lastReadTime === null
```

Comment:
You can doublecheck this, but I am pretty sure this is the "bug" I fixed for Kavitha. If lastReadData.lastReadTime is null, I don't think lastReadTime will be null, I think it will just be an invalid date. You could check getTime() isNan().

But the best approach is probably:

  const lastReadTime = lastReadData.lastReadTime ? new Date(lastReadData.lastReadTime as string) : null;